pkg/scheduler/factory_test.go

```diff
@@ -553,6 +554,39 @@ func TestDefaultErrorFunc_NodeNotFound(t *testing.T) {
 	}
 }
 
+func TestDefaultErrorFunc_PodAlreadyBound(t *testing.T) {
+	stopCh := make(chan struct{})
+	defer close(stopCh)
+
+	nodes := []v1.Node{{ObjectMeta: metav1.ObjectMeta{Name: "foo"}}}
+
+	testPod := &v1.Pod{ObjectMeta: metav1.ObjectMeta{Name: "test-pod", Namespace: "default"}, Spec: v1.PodSpec{NodeName: "foo"}}
+
+	client := fake.NewSimpleClientset(&v1.PodList{Items: []v1.Pod{*testPod}}, &v1.NodeList{Items: nodes})
+	informerFactory := informers.NewSharedInformerFactory(client, 0)
+	podInformer := informerFactory.Core().V1().Pods()
+	// Need to add testPod to the store.
+	podInformer.Informer().GetStore().Add(testPod)
+
+	queue := internalqueue.NewPriorityQueue(nil, internalqueue.WithClock(clock.NewFakeClock(time.Now())))
+	schedulerCache := internalcache.New(30*time.Second, stopCh)
+
+	for i := range nodes {
+		node := nodes[i]
```

Comment:
you don't need a list here, there is only one node.

Reply:
i have fixed it